micrometer-metrics-examples/README.adoc

```diff
@@ -0,0 +1,116 @@
+= Vert.x Micrometer Metrics examples
+
+This project shows simple examples using the Vert.x Micrometer Metrics module.
+
+There are examples for _Prometheus_, _InfluxDB_ and _JMX_ backends.
+Each of them deploys 3 verticles:
+
+* `SimpleWebServer`, which creates an HTTP server - visit http://localhost:8080/ to trigger requests.
+* `EventbusProducer`, which sends messages over the event bus
+* `EventbusConsumer`, which receives event bus messages
+
+Each of them simulates a random processing time.
+
+For _Prometheus_, the verticle `WebServerForBoundPrometheus` is an alternative of `SimpleWebserver`
+that allows to bind the prometheus `/metrics` endpoint to an existing HTTP server.
+
+== Launching the example from the command line
+
+First, build the fat jar:
+```bash
+mvn clean package
+```
+
+Then run any of these Main classes:
+
+=== Prometheus
+
+Note: link:https://prometheus.io/docs/prometheus/latest/getting_started/[check here] for Prometheus getting started guide.
+
+```bash
+java -cp target/micrometer-metrics-examples-3.5.1-fat.jar io.vertx.example.micrometer.prometheus.Main
+```
+
+The prometheus server must be configured with a scrape configuration to localhost:8081
+
+```yaml
+  - job_name: 'vertx-8081'
+    static_configs:
+      - targets: ['localhost:8081']
+```
+
+By default, histogram sampling will not be provided.
+It can be manually turned on in the Micrometer registry, as shown in commented code in class
+`io.vertx.example.micrometer.prometheus.Main`:
+
+```java
+registry.config().meterFilter(
+    new MeterFilter() {
+      @Override
+      public DistributionStatisticConfig configure(Meter.Id id, DistributionStatisticConfig config) {
+        return DistributionStatisticConfig.builder()
+            .percentilesHistogram(true)
+            .build()
+            .merge(config);
+      }
+    });
+```
+
+See also link:https://micrometer.io/docs/concepts#_histograms_and_percentiles[Micrometer documentation].
+
+=== Prometheus with endpoint bound to existing server
+
+```bash
+java -cp target/micrometer-metrics-examples-3.5.1-fat.jar io.vertx.example.micrometer.ometheus.MainWithBoundPrometheus
+```
+
+The prometheus server must be configured with a scrape configuration to localhost:8080
+
+```yaml
+  - job_name: 'vertx-8080'
+    static_configs:
+      - targets: ['localhost:8080']
+```
+
+=== InfluxDB
+
+This sample application expects an InfluxDB server running on localhost, port 8086, without authentication.
+For quick setup, this docker command will run it:
```

Comment:
_you can run it this docker command_

Reply:
Ok, I was trying to use the passive form as much as possible but maybe it doesn't sound too good